model/vfs/vfsswift/swift.go

```diff
@@ -36,7 +37,21 @@ func DeleteContainer(c *swift.Connection, container string) error {
 			return err
 		}
 	}
-	return c.ContainerDelete(container)
+
+	// XXX Swift has told us that all the files have been deleted on the bulk
+	// delete, but it only means that they have been deleted on one object
+	// server (at least). And, when we try to delete the container, Swift can
+	// send an error as some container servers will still have objects
+	// registered for this container. We will try several times to delete the
+	// container to work-around this limitation.
+	for i := 0; i < 5; i++ {
+		err = c.ContainerDelete(container)
+		if err == nil {
+			return nil
+		}
+		time.Sleep(5 * time.Second)
```

Comment:
Wouldn't it be better to move sleep to the beginning of the `for` loop inside an `if i != 0 { }` ?
This would have the benefit to fail immediately on last turn instead of waiting 5 seconds after last fail to return.

Reply:
But it has the con of always waiting 5 seconds before the first try.

Reply:
Not if put I side àn `if i != 0 {}`